internal/fsutil/fsutil.go

```
@@ -0,0 +1,74 @@
+// the same directory, fsyncs it, applies mode, and renames it over path. A
+// crash mid-write leaves any existing file untouched. mode is applied with
+// os.Chmod, so it is exact (not filtered by umask); pass the original file's
+// mode to preserve it across replacements.
+func WriteFileAtomic(path string, data []byte, mode os.FileMode) error {
+	tmp, err := os.CreateTemp(filepath.Dir(path), "."+filepath.Base(path)+".tmp.*")
+	if err != nil {
+		return fmt.Errorf("create tmp: %w", err)
+	}
+	tmpPath := tmp.Name()
+	if _, err := tmp.Write(data); err != nil {
+		tmp.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `tmp.Close` is not checked

(errcheck)